perf/reader.go

```diff
@@ -341,6 +340,30 @@ func (pr *Reader) Read() (Record, error) {
 	}
 }
 
+// Mute stops all notifications from this perf reader.
+//
+// Subsequent calls to Read will block until a call to Unmute.
+func (pr *Reader) Mute() error {
+	for i := 0; i < len(pr.rings); i++ {
```

Comment:
Regards handling failures, I could see arguments for either:
* Collecting all failures and returning a slice of errors, or
* Walking back through the successful operations here and undoing them (which could also technically fail...)

In Cilium we ignore the errors in `Mute()` and return one back like this in `Unmute()`. I did one extra enhancement in this version where I check against `ENOENT`, which also has the nice property of making `Mute()` idempotent without any additional fields to hold mute/unmute state.

Happy to take direction here.

Reply:
Slices of error are always difficult to use IMO, I think you made the right call here. Undoing also doesn't seem like it would help much. What does Cilium currently do if Mute / Unmute fail? Does it keep on running or does it bail out?